### PR TITLE
PowerSubsystem and PowerSupply schema Refresh for MEX support

### PR DESCRIPTION
**Pull request by lkammath**
This PR has latest refresh commit 
https://gerrit.openbmc-project.xyz/c/openbmc/bmcweb/+/40326

1) PowerSubsystem and PowerSupply Schema updates for MEX chassis
2) Skip erroring out part based on EC

### CODE REVIEW

**Comment by lkammath** [comment]
Not ready for review. Will add reviewers once i confirm everything is good

File: redfish-core/include/redfish.hpp [hunk 1]
```diff
@@ -82,8 +82,11 @@ class RedfishService
         requestRoutesThermal(app);
         requestRoutesPower(app);
 #endif
-        requestRoutesThermalSubsystem(app);
-        requestRoutesThermalMetrics(app);
```
Comment (gtmills) [inline redfish-core/include/redfish.hpp:86]: It looks like this drops ThermalSubsystem? Why the changes here?

File: redfish-core/include/registries/privilege_registry.hpp [hunk 2]
```diff
@@ -1421,6 +1421,30 @@ const static auto& putPowerDistributionMetrics =
 const static auto& deletePowerDistributionMetrics =
     privilegeSetConfigureComponents;
 
+// PowerSubsystem
+const static auto& getPowerSubsystem = privilegeSetLogin;
```
Comment (gtmills) [inline redfish-core/include/registries/privilege_registry.hpp:1425]: Why the changes here?

File: redfish-core/lib/power_subsystem.hpp [hunk 3]
```diff
@@ -28,7 +29,7 @@ inline void
 inline void requestRoutesPowerSubsystem(App& app)
 {
     BMCWEB_ROUTE(app, "/redfish/v1/Chassis/<str>/PowerSubsystem/")
-        .privileges({{"Login"}})
+        .privileges(redfish::privileges::getPowerSubsystem)
```
Comment (gtmills) [inline redfish-core/lib/power_subsystem.hpp:32]: Why the change here?

File: redfish-core/include/redfish.hpp (hunk 1 above)
Comment (ChicagoDuan) [reply redfish-core/include/redfish.hpp:86]: We should not remove the ThermalSubsystem and ThermalMetrics.

File: redfish-core/include/registries/privilege_registry.hpp (hunk 2 above)
Comment (ChicagoDuan) [reply redfish-core/include/registries/privilege_registry.hpp:1425]: Done. I deleted it. @lkammath  Please pick the latest patchset.

File: redfish-core/lib/power_subsystem.hpp (hunk 3 above)
Comment (ChicagoDuan) [reply redfish-core/lib/power_subsystem.hpp:32]: Done. I deleted it and add a Todo. @lkammath Please pick the latest patchset.

File: redfish-core/include/redfish.hpp [hunk 4]
```diff
@@ -84,6 +84,11 @@ class RedfishService
 #endif
         requestRoutesThermalSubsystem(app);
         requestRoutesThermalMetrics(app);
+#ifdef BMCWEB_NEW_POWERSUBSYSTEM_THERMALSUBSYSTEM
+        requestRoutesPowerSubsystem(app);
+        requestRoutesPowerSupplyCollection(app);
+        requestRoutesPowerSupply(app);
```
Comment (gtmills) [inline redfish-core/include/redfish.hpp:90]: Look at line 99 - 101, these are already here. 
I am fine if you move to the ifdef but then need to remove the old ones 

If you want 

 #ifdef BMCWEB_NEW_POWERSUBSYSTEM_THERMALSUBSYSTEM
         requestRoutesThermalSubsystem(app);
        requestRoutesThermalMetrics(app);
         requestRoutesPowerSubsystem(app);
        requestRoutesPowerSupplyCollection(app);
        requestRoutesPowerSupply(app);
  #endif
  
  or just leave this file alone

**Review by gtmills — CHANGES_REQUESTED** [review]
Please find the requests

File: static/redfish/v1/JsonSchemas/PowerSubsystem/PowerSubsystem.json [hunk 5]
```diff
@@ -211,4 +211,4 @@
     "owningEntity": "DMTF",
     "release": "2021.2",
     "title": "#PowerSubsystem.v1_1_0.PowerSubsystem"
-}
```
Comment (ChicagoDuan) [inline static/redfish/v1/JsonSchemas/PowerSubsystem/PowerSubsystem.json:214]: Why delete it and add it again?

File: static/redfish/v1/JsonSchemas/PowerSupply/PowerSupply.json [hunk 6]
```diff
@@ -555,4 +555,4 @@
     "owningEntity": "DMTF",
     "release": "2021.1",
     "title": "#PowerSupply.v1_1_0.PowerSupply"
-}
```
Comment (ChicagoDuan) [inline static/redfish/v1/JsonSchemas/PowerSupply/PowerSupply.json:558]: Why delete it and add it again?

File: static/redfish/v1/JsonSchemas/index.json [hunk 7]
```diff
@@ -403,4 +403,4 @@
       "@odata.id": "/redfish/v1/JsonSchemas/redfish-schema"
     }
   ]
-}
```
Comment (ChicagoDuan) [inline static/redfish/v1/JsonSchemas/index.json:406]: Why delete it and add it again?

File: static/redfish/v1/JsonSchemas/PowerSupply/PowerSupply.json (hunk 6 above)
Comment (lkammath) [reply static/redfish/v1/JsonSchemas/PowerSupply/PowerSupply.json:558]: I don't know why it is showing really.. but , everything seems be OK to me..

File: static/redfish/v1/JsonSchemas/PowerSubsystem/PowerSubsystem.json (hunk 5 above)
Comment (lkammath) [reply static/redfish/v1/JsonSchemas/PowerSubsystem/PowerSubsystem.json:214]: I don't know why it is showing really.. but , everything seems be OK to me..

File: static/redfish/v1/JsonSchemas/index.json (hunk 7 above)
Comment (lkammath) [reply static/redfish/v1/JsonSchemas/index.json:406]: I don't know why it is showing really.. but , everything seems be OK to me..

**Review by gtmills — APPROVED** [review]
